sdk/search/azure-search-documents/src/samples/java/com/azure/search/documents/indexes/CreateIndexWithFieldBuilderExample.java

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.indexes;
+
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.util.Configuration;
+import com.azure.search.documents.indexes.models.FieldBuilderOptions;
+import com.azure.search.documents.indexes.models.SearchField;
+import com.azure.search.documents.indexes.models.SearchIndex;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class CreateIndexWithFieldBuilderExample {
+    /**
+     * From the Azure portal, get your Azure Cognitive Search service name and API key and populate ADMIN_KEY and
+     * SEARCH_SERVICE_NAME.
+     */
+    private static final String ENDPOINT = Configuration.getGlobalConfiguration().get("AZURE_COGNITIVE_SEARCH_ENDPOINT");
+    private static final String ADMIN_KEY = Configuration.getGlobalConfiguration().get("AZURE_COGNITIVE_SEARCH_API_KEY");
+
+    public static void main(String[] args) {
+        // Create the SearchIndex client.
+        SearchIndexClient client = new SearchIndexClientBuilder()
+            .endpoint(ENDPOINT)
+            .credential(new AzureKeyCredential(ADMIN_KEY))
+            .buildClient();
+
+        // Use the SearchIndexClient to create SearchFields from your own model that has fields or methods annotated
+        // with @SimpleField or @SearchableField.
+        List<SearchField> indexFields = SearchIndexClient.buildSearchFields(Hotel.class, new FieldBuilderOptions());
+        String indexName = "hotels";
+        SearchIndex newIndex = new SearchIndex(indexName, indexFields);
+
+        // Create index.
+        client.createIndex(newIndex);
+
+        // Cleanup index resource.
+        client.deleteIndex(indexName);
+    }
+
+    /**
+     * A hotel.
+     */
+    public class Hotel {
+        @SimpleField(isKey = true, isFilterable = true, isSortable = true)
```

Comment:
I am assuming the difference between simple and searchable is that one if being indexed and the other is not?

If simple field is not indexed, how is it filterable? Can we search non-indexed fields. If so, I am assuming the performance will be super slow for them?

Reply:
Difference between a `SearchableField` and a `SimpleField` is just that a searchable field is searchable while a simple field isn't